### ISSUE TITLE
Bug: Rendering Issue HardPhong/SoftPhong Shader when UV verts of texture > Mesh verts

### ISSUE THREAD

**Issue by vpolflie** [facebookresearch/pytorch3d#372]
When rendering any mesh  where a 3D vertex might have multiple UV coordinates the renderer will get a CUDA device side error since it is trying to access data that isn't correctly allocated.

The rendering breaks on line 806 or textures.py:

`faces_verts_uvs = verts_uvs[faces_uvs]`

This is because even though my starting mesh has for example the following allocated tensors:

```
verts_uvs= 900x2
faces_uvs=300x3
verts_mesh=600x3
faces_mesh=300x3
```

When we get to this line of code, it executes the equivalent of:

`faces_verts_uvs = tensor_of_length_600[tensor_of_length_300_containing_indices_ranging_to_900]`

So how come `verts_uvs` only has 600 elements instead of the aforementioned 900?

Well I think the issue lies with the following line in textures.py at line 738:

```
self._verts_uvs_list = padded_to_list(
                    self._verts_uvs_padded, split_size=self._num_verts_per_mesh
                )
```

This function calls `def padded_to_list(x: torch.Tensor, split_size: Union[list, tuple, None] = None):` at line 58 of utils.py where the `self._verts_uvs_padded` is transformed into the padded representation. However since `self._num_verts_per_mesh` reference to the amount of mesh vertices instead of the amount of UV vertices this function returns a tensor of shape 600x2 instead of the expected 900x2.

**Comment by vpolflie** [comment]
Fixed some numbers in my initial version

**Comment by vpolflie** [comment]
Note this only seems to happen when using `mesh.extend(batch_size)`

**Comment by bottler** [comment]
I think this was fixed in 90f6a005 and the fix is included in the latest release 0.2.5 of pytorch3d.

**Comment by vpolflie** [comment]
My version is still 0.2.0, I'll update and check whether it works right now. Thanks for the update. I'll close the issue meanwhile.
